motionplan/rrtStarConnect.go

```diff
@@ -176,64 +212,111 @@ func (mp *rrtStarConnectMotionPlanner) rrtBackgroundRunner(ctx context.Context,
 		}
 
 		// get next sample, switch map pointers
-		target = referenceframe.RandomFrameInputs(mp.frame, mp.randseed)
+		target, err = mp.sample(map1reached, i)
+		if err != nil {
+			rrt.solutionChan <- &rrtPlanReturn{planerr: err, maps: rrt.maps}
+			return
+		}
+		map1, map2 = map2, map1
 	}
 	mp.logger.Debug("RRT* exceeded max iter")
 	rrt.solutionChan <- shortestPath(rrt.maps, shared)
 }
 
+func (mp *rrtStarConnectMotionPlanner) sample(rSeed node, sampleNum int) ([]referenceframe.Input, error) {
```

Comment:
If this is the same as the one in cbirrt we could think about moving it into rrt.go and using it for both. But this is optional